public/scripts/project/lock.js

```diff
@@ -0,0 +1,51 @@
+// Our lock system for projects is very simple. If the lock exists,
+// it's almost guaranteed to be valid since the expiry serves
+// to prevent completely locking access to a project in edge cases.
+define(function() {
+  var EXPIRY = 60000; // 1000ms * 60s
+  var LOCK_PREFIX = "bramble-lock-";
+
+  var getItem = localStorage.getItem.bind(localStorage);
+  var setItem = localStorage.setItem.bind(localStorage);
+  var removeItem = localStorage.removeItem.bind(localStorage);
+
+  // Sets the lock in localStorage, or returns false
+  // if the lock already exists and is valid
+  function setLock(projectID) {
+    function reacquire() {
+      setTimeout(reacquire, EXPIRY);
+      setItem(LOCK_PREFIX + projectID, Date.now().toString());
+      return true;
+    }
+
+    var lock = getLock(projectID);
```

Comment:
In fact, write a function that does this automatically for you:
```js
function projectLockID(projectID) {
  return LOCK_PREFIX + projectID;
}
```
and call it everywhere instead of doing this manually.  And/or wrap your calls to `localStorage.*` functions to automatically do this for you.